libraries/freertos_plus/standard/https/src/private/iot_https_internal.h

```diff
@@ -0,0 +1,276 @@
+    PROCESSING_STATE_FINISHED,                  /**< @brief Filling and parsing of both buffers is finished. */
+    PROCESSING_STATE_SEARCHING_HEADER_BUFFER    /**< @brief The header buffer is being search. */
+} IotHttpsResponseBufferState_t;
+
+/*-----------------------------------------------------------*/ 
+
+/**
+ * @brief Represents an HTTP connection.
+ */
+typedef struct _httpsConnection
+{
+    IotNetworkInterface_t * pNetworkInterface;  /**< @brief Network interface with calls for connect, disconnect, send, and receive. */
+    void *pNetworkConnection;                   /**< @brief Pointer to the network connection to use pNetworkInterface calls on. */
+    IotSemaphore_t connSem;                     /**< @brief Binary semaphore to lock the connection if a request/response is currently in progress. */
+    bool nonPersistent;                         /**< @brief Non persistent flag to indicate closing the connection immediately after receiving the request. */
+    IotSemaphore_t rxStartSem;                  /**< @brief Semaphore indicating that data on the network is ready to read. */
+    IotSemaphore_t rxFinishSem;                 /**< @brief Semaphore indicating that the data on the network is done being read. */
+    uint32_t timeout;                           /**< @brief Timeout for a connection and waiting for a response from the network. */
+    bool isConnected;                           /**< @brief true if a connection was successful most recently on this context. We have not way of knowing if the 
+                                                            serving closed the connection on us because that error is unique to the underlying TLS layer. This is set
+                                                            to false initially, set to true for a successful intentional call to connect, and then set to false only
+                                                            after an explicit disconnect with a non-persistent request or a call to @ref https_client_function_disconnect. */
+} _httpsConnection_t;
+
+/**
+ * @brief Represents an HTTP response.
+ */
+typedef struct _httpsResponse
+{
+    uint8_t * pHeaders;      /**< @brief Pointer to the start of the headers buffer. */
+    uint8_t * pHeadersEnd;   /**< @brief Pointer to the end of the headers buffer. */
+    uint8_t * pHeadersCur;   /**< @brief Pointer to the next location to fill in the headers buffer. */
+    uint8_t * pBody;         /**< @brief Pointer to the start of the body buffer. */
+    uint8_t * pBodyCur;      /**< @brief Pointer to the next location to fill in the body buffer. */
+    uint8_t * pBodyEnd;      /**< @brief Pointer to the end of the body buffer. */
+    http_parser httpParser; /**< @brief Instance of the http_parser for response processing. */
+    uint16_t status;        /**< @brief The HTTP response status code of this response. */
+    IotHttpsMethod_t method;    /**< @brief The method of the originating request. */
```

Comment:
```pReqHandle``` is part of this structure. Do we need the ```method``` here ?

Reply:
No I suppose we don't.